### DISCUSSION TITLE
How do you add friction to an object?

### DISCUSSION THREAD

**Discussion by latticeyu** [DualSPHysics/DualSPHysics#264] [DUALSPHYSICS_V5.2]
In DualSPHysics, how do you add a coefficient of friction to a certain face of a block to give it frictional force? I would appreciate it if you could answer this question.

**Comment by imestevez** [comment]
Material properties are added to an `mk`, so you have to define the face with its own `mk`, and then define its material properties following the DualSPHysics-Chrono template:

https://github.com/DualSPHysics/DualSPHysics/blob/ecf3e9a98fcc3957840dd11b9973c5c60e134065/doc/xml_format/_FmtXML_Chrono.xml#L10-L28

For instance, to define the properties of PVC:
1) Add the XML that contains the material properties: `<propertyfile file="Floating_Materials.xml" path="materials" />`
2) Define the material properties to an object:
      - For a floating body, option `<floating mkbound="X" relativeweight="1.0" property="pvc" />`
      - For a fixed body, option `<link mkbound="0" property="pvc" comment="Property for the fixed bound" /> `


In addition, you can download the fullpackage of DualSPHysics (https://dual.sphysics.org/) and have a look to at examples/chrono/07_DamBreakCubes

**Comment by latticeyu** [comment]
thank you for your help very much



---Original---
From: "Iván Martínez ***@***.***&gt;
Date: Mon, Nov 25, 2024 23:37 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [DualSPHysics/DualSPHysics] How do you add friction to an object?(Discussion #264)




 
Material properties are added to an mk, so you have to define the face with its own mk, and then define its material properties following the DualSPHysics-Chrono template:
 
https://github.com/DualSPHysics/DualSPHysics/blob/ecf3e9a98fcc3957840dd11b9973c5c60e134065/doc/xml_format/_FmtXML_Chrono.xml#L10-L28
 
For instance, to define the properties of PVC:
  
Add the XML that contains the material properties: <propertyfile file="Floating_Materials.xml" path="materials" /&gt;
 
Define the material properties to an object:  
For a floating body, option <floating mkbound="X" relativeweight="1.0" property="pvc" /&gt;
 
For a fixed body, option <link mkbound="0" property="pvc" comment="Property for the fixed bound" /&gt; 
  
  
In addition, you can download the fullpackage of DualSPHysics (https://dual.sphysics.org/) and have a look to at examples/chrono/07_DamBreakCubes
